agent/http-server.c

```diff
@@ -69,6 +72,52 @@ hwangsae_http_server_set_recording_dir (HwangsaeHttpServer * server,
   server->recording_dir = g_strdup (recording_dir);
 }
 
+static gchar *
+_get_local_ip (void)
+{
+  struct ifaddrs *addrs;
+  struct ifaddrs *it;
+  gchar *result = NULL;
+
+  if (getifaddrs (&addrs) < 0) {
+    return NULL;
+  }
+
+  for (it = addrs; !result && it; it = it->ifa_next) {
+    socklen_t addr_len;
+    char buf[INET6_ADDRSTRLEN + 1];
+
+    /* Ignore interfaces that are down, not running or don't have an IP. We also
+     * want to skip loopbacks. */
+    if ((it->ifa_flags & (IFF_UP | IFF_RUNNING | IFF_LOOPBACK)) !=
+        (IFF_UP | IFF_RUNNING) || it->ifa_addr == NULL) {
+      continue;
+    }
+
+    switch (it->ifa_addr->sa_family) {
+      case AF_INET:
+        addr_len = sizeof (struct sockaddr_in);
+        break;
+      case AF_INET6:
+        addr_len = sizeof (struct sockaddr_in6);
+        break;
+      default:
+        continue;
+    }
+
+    if (getnameinfo (it->ifa_addr, addr_len, buf, sizeof (buf), NULL, 0,
+            NI_NUMERICHOST) != 0) {
+      continue;
+    }
+
+    result = g_strdup (buf);
+  }
+
+  freeifaddrs (addrs);
+
+  return result;
+}
+
```

Comment:
The function looks like the one from `HwangsaeRelay`. What about creating common `hwangsae_utils_get_local_ip()` instead of duplicating code? (Move the function to a new file in `hwangsae` dir.)

Reply:
Yes, I wanted to make it simpler, as it was just one function, but I would really love to do that. Any other function you think we can move there?

Reply:
Nothing else comes to mind in Hwangsae. Let's start with just `_get_local_ip()`.

Reply:
Great, thanks for you review.